Kaptain 2.0.0 Docs sdk folder (#4319)
mesosphere/dcos-docs-site: pages/dkp/kaptain/2.0.0/sdk/1.0.x/credentials/index.md

---
layout: layout.pug
navigationTitle: Accessing Docker and Cloud Storage
title: Configuring Access to Docker and Cloud Storage
beta: false
menuWeight: 2
---

This tutorial shows how to configure access to Docker and Cloud Storage.

When the model is trained, all the model files are packed into a Docker
image, which is then used for the training itself, and for hyperparameter
tuning later on. In order to build that image, the SDK must be provided
with Docker credentials, so that it can publish the resulting images
to the registry specified in `Model.image` attribute.

As the image building happens on the cluster, the model files are first
uploaded to a blob storage such as S3, GCS, or MinIO and then used by
the builder. By default, the SDK uses a cluster-local MinIO installation

> which doesn't require any additional configuration. If users wish to

Author: ```suggestion
which does not require any additional configuration. If users wish to
```

dialogue, create a file named `pod_default.yaml` with the following contents:

    apiVersion: "kubeflow.org/v1alpha1"
    kind: PodDefault
    metadata:
      name: docker-config
      namespace: user
    spec:
      selector:
        matchLabels:
          docker-config: "true"
      desc: "Add Docker registry credentials"
      volumeMounts:
        - name: docker-secret-volume
          mountPath: /home/kubeflow/.docker/
      volumes:
        - name: docker-secret-volume
          secret:
            secretName: docker-secret

<p class="message--warning">
<strong>WARNING: </strong>
The volume name and <code>mountPath</code> must be unique across all <code>PodDefault</code>
objects to avoid conflicts when mounting <code>Secrets</code> to <code>Pods</code>.
</p>

Create the `PodDefault` resource from the file using the following command:

    kubectl apply -f pod_default.yaml

After that, the Docker credentials secret will be available for
selection in the Notebook Spawner UI and, if selected, will be mounted
to `/home/kubeflow/.docker/`:

![image](img/docker_registry_secret.png)

## AWS credentials

### File-based and environment variable based configuration

> There are two ways to make AWS credentials available in a notebook:

Author: @gvasquezvargas

> There are two ways to make AWS credentials available in a notebook:

Reviewer: ```suggestion
You can make AWS credentials available in a notebook in two different ways:
```

* As a configuration file mounted to the `Pod` from a `Secret`
* As environment variables injected into the `Pod` from a `Secret`

<p class="message--note">
<strong>NOTE: </strong>
Which method to use depends on what AWS settings you need to configure
<ul>

> <li>The configuration file method is recommended when working with the default account settings, i.e. when only credentials

Author: ```suggestion
<li>The configuration file method is recommended when working with the default account settings, that is when only credentials
```
